src/Microsoft.AspNetCore.Razor.Language/Extensions/FunctionsDirectivePass.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.Extensions
+{
+    public class FunctionsDirectivePass : RazorIRPassBase, IRazorDirectiveClassifierPass
```

Comment:
We should pick an approach of static XYZDirective class with a static register method and a nested private class (that's what `InjectDirective` does as an example) or what you have in this PR, static class, register, the directive class and a public pass. 

I worry about exposing the Functions pass here. Is there value in letting users see this? Same questions for inject.